x-pack/plugin/eql/qa/ccs-rolling-upgrade/src/test/java/org/elasticsearch/xpack/eql/qa/ccs_rolling_upgrade/EqlCcsRollingUpgradeIT.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    static RestHighLevelClient newRemoteClient() {
+        return new RestHighLevelClient(RestClient.builder(randomFrom(parseHosts("tests.rest.remote_cluster"))));
+    }
+
+    static int indexDocs(RestHighLevelClient client, String index, int numDocs) throws IOException {
+        for (int i = 0; i < numDocs; i++) {
+            client.index(new IndexRequest(index).id("id_" + i).source("f", i, "@timestamp", i), RequestOptions.DEFAULT);
+        }
+
+        refresh(client.getLowLevelClient(), index);
+        return numDocs;
+    }
+
+    void verify(String localIndex, int localNumDocs, String remoteIndex, int remoteNumDocs) {
+        try (RestClient localClient = newLocalClient().getLowLevelClient()) {
+
+            Request request = new Request("POST", "/" + randomFrom(remoteIndex, localIndex + "," + remoteIndex) + "/_eql/search");
+            int size = between(1, 100);
+            int id1 = between(0, 5);
+            int id2 = between(6, Math.min(localNumDocs - 1, remoteNumDocs - 1));
+            request.setJsonEntity(
+                "{\"query\": \"sequence [any where f == " + id1 + "] [any where f == " + id2 + "] \", \"size\": " + size + "}"
+            );
+            Response response = localClient.performRequest(request);
+            String responseText = EntityUtils.toString(response.getEntity());
+            assertTrue(responseText.contains("\"sequences\":[{"));
+            assertTrue(responseText.contains("\"_id\":\"id_" + id1 + "\""));
+            assertTrue(responseText.contains("\"_id\":\"id_" + id2 + "\""));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public void testSequences() throws Exception {
+        String localIndex = "test_bwc_search_states_index";
+        String remoteIndex = "test_bwc_search_states_remote_index";
+        try (RestHighLevelClient localClient = newLocalClient(); RestHighLevelClient remoteClient = newRemoteClient()) {
+            createIndex(
+                localClient.getLowLevelClient(),
+                localIndex,
+                Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, between(1, 5)).build(),
+                "{\"properties\": {\"@timestamp\": {\"type\": \"date\"}}}",
+                null
+            );
+            int localNumDocs = indexDocs(localClient, localIndex, between(10, 100));
+            createIndex(
+                remoteClient.getLowLevelClient(),
+                remoteIndex,
+                Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, between(1, 5)).build(),
+                "{\"properties\": {\"@timestamp\": {\"type\": \"date\"}}}",
+                null
+            );
+            int remoteNumDocs = indexDocs(remoteClient, remoteIndex, between(10, 100));
+
+            configureRemoteClusters(getNodes(remoteClient.getLowLevelClient()));
+            int iterations = between(1, 20);
+            for (int i = 0; i < iterations; i++) {
```

Comment:
Curious why we went for running it multiple times.

Reply:
The dataset and the queries are randomized, this repetition just adds some more robustness to the test. Anyway, nothing critical. Also from the performance point of view it's negligible, most of the cost of this test is the cluster startup, not the test itself.